frontend/src/GroceryList.js

```diff
@@ -8,15 +8,25 @@ import DeleteIcon from '@material-ui/icons/Delete';
 import ExpandMoreRoundedIcon from '@material-ui/icons/ExpandMoreRounded';
 import {List, ListItem} from '@material-ui/core';
 import {withStyles} from '@material-ui/core';
+import {Select, MenuItem} from "@material-ui/core";
 import { withTheme } from "@material-ui/styles";
 import { color } from "chart.js/helpers";
 
 import TagManager from 'react-gtm-module'
+import { ContactSupport } from "@material-ui/icons";
 
 
 const GHGI_API_ADDRESS = 'https://api.sexytofu.org/api.ghgi.org:443';
 const NATIVE_API_ADDRESS =  process.env.API_HOST || "http://localhost:8000";
-
+// Matches all GHGI units from https://github.com/ghgindex/ghgi/blob/main/ghgi/parser.py
+// All units we support
+const ALL_UNITS = ['milliliter', 'liter', 'gram', 'kilogram', 'cup', 
+                    'tablespoon', 'teaspoon', 'ounce', 'pound', 'quart', 
+                    'pint', 'dash', 'pinch', 'handful', 'fistful', 'smidgen', 'ea'];
+// Limited selection of units visible for sake of dropdown menu
```

Comment:
I didn't add everything yet, but ALL_UNITS is based on units GHGI handles: https://github.com/ghgindex/ghgi/blob/main/ghgi/parser.py#L9

ALL_UNITS is everything you can type in, while VISIBLE_UNITS is the units you can select from the dropdown selection. It's possible to input more units than selectable by typing it in as the initial food search input.

Reply:
Alphabetical order is ok, but it would be nice to keep similar units grouped together (eg. cup, pint, quart // tablespoon, teaspoon).

Reply:
The parser handles standardization including plural forms, abbreviations, etc., so it's pretty nice. For ALL_UNITS, let's only use the values of the key-value mappings (the "UNITS" var): e.g. "kg" rather than "kilogram".

Reply:
If we use Autocomplete, then I am not too worried about alphabetical order or list being too long, so we can group similar units together like you suggested